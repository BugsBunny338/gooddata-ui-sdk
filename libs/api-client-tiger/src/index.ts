// (C) 2019-2020 GoodData Corporation
import { AxiosInstance } from "axios";
import { tigerExecutionClientFactory } from "./execution";
import {
    tigerLabelElementsClientFactory,
    LabelElementsConfiguration,
    LabelElementsConfigurationParameters,
    LabelElementsBaseApi,
    LabelElementsRequestArgs,
} from "./labelElements";
import {
    tigerMetadataClientFactory,
    MetadataConfiguration,
    MetadataConfigurationParameters,
    MetadataBaseApi,
    MetadataRequestArgs,
} from "./metadata";
import { tigerValidObjectsClientFactory } from "./validObjects";
import { axios as defaultAxios, newAxios } from "./axios";

export { VisualizationObject } from "./gd-tiger-model/VisualizationObject";
export * from "./gd-tiger-model/typeGuards";

export { newAxios };

export * from "./generated/afm-rest-api/api";
export * from "./generated/metadata-json-api/api";

export {
    tigerMetadataClientFactory,
    MetadataConfiguration,
    MetadataConfigurationParameters,
    MetadataBaseApi,
    MetadataRequestArgs,
    tigerExecutionClientFactory,
    tigerLabelElementsClientFactory,
    LabelElementsConfiguration,
    LabelElementsConfigurationParameters,
    LabelElementsBaseApi,
    LabelElementsRequestArgs,
    tigerValidObjectsClientFactory,
};

export interface ITigerClient {
    metadata: ReturnType<typeof tigerMetadataClientFactory>;
    execution: ReturnType<typeof tigerExecutionClientFactory>;
    labelElements: ReturnType<typeof tigerLabelElementsClientFactory>;
    validObjects: ReturnType<typeof tigerValidObjectsClientFactory>;
}

/**
 * Tiger execution client
 *
 */
export const tigerClientFactory = (axios: AxiosInstance): ITigerClient => {
    const execution = tigerExecutionClientFactory(axios);
    const labelElements = tigerLabelElementsClientFactory(axios);
    const metadata = tigerMetadataClientFactory(axios);
    const validObjects = tigerValidObjectsClientFactory(axios);

    return {
        execution,
        labelElements,
        metadata,
        validObjects,
    };
};

const defaultTigerClient = tigerClientFactory(defaultAxios);

export default defaultTigerClient;
