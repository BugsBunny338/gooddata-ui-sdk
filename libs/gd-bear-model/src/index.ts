// (C) 2007-2019 GoodData Corporation

export { GdcExecuteAFM } from "./executeAfm/GdcExecuteAFM";
export { GdcExecution } from "./executeAfm/GdcExecution";
export { GdcExtendedDateFilters } from "./extendedDateFilters/GdcExtendedDateFilters";
export { GdcVisualizationObject } from "./visualizationObject/GdcVisualizationObject";
export { GdcVisualizationClass } from "./visualizationObject/GdcVisualizationClass";
export { GdcDashboardExport } from "./dashboard/GdcDashboardExport";
export { GdcDashboardLayout } from "./dashboard/DashboardLayout";
export { GdcCatalog } from "./catalog/GdcCatalog";
export { GdcMetadata } from "./meta/GdcMetadata";

export { sanitizeDateFilters } from "./dashboard/utils";
export { getAttributesDisplayForms } from "./visualizationObject/utils";
