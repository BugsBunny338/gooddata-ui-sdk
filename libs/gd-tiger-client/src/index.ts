// (C) 2019-2020 GoodData Corporation
import { AxiosInstance } from "axios";
import { tigerExecutionClientFactory } from "./execution";
import { tigerLabelElementsClientFactory } from "./labelElements";
import { tigerMetadataClientFactory } from "./metadata";
import { tigerValidObjectsClientFactory } from "./validObjects";
import { axios as defaultAxios, newAxios } from "./axios";

export { ExecuteAFM } from "./gd-tiger-model/ExecuteAFM";
export { Execution } from "./gd-tiger-model/Execution";

export { newAxios };

export * from "./generated/afm-rest-api/api";
export * from "./generated/metadata-json-api/api";

export interface ITigerClient {
    metadata: ReturnType<typeof tigerMetadataClientFactory>;
    execution: ReturnType<typeof tigerExecutionClientFactory>;
    labelElements: ReturnType<typeof tigerLabelElementsClientFactory>;
    validObjects: ReturnType<typeof tigerValidObjectsClientFactory>;
}

/**
 * Tiger execution client
 *
 */
export const tigerClientFactory = (axios: AxiosInstance): ITigerClient => {
    const execution = tigerExecutionClientFactory(axios);
    const labelElements = tigerLabelElementsClientFactory(axios);
    const metadata = tigerMetadataClientFactory(axios);
    const validObjects = tigerValidObjectsClientFactory(axios);

    return {
        execution,
        labelElements,
        metadata,
        validObjects,
    };
};

const defaultTigerClient = tigerClientFactory(defaultAxios);

export default defaultTigerClient;
