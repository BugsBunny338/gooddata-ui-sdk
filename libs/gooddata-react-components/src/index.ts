// (C) 2007-2019 GoodData Corporation
import * as VisEvents from "./interfaces/Events";
import CatalogHelper from "./helpers/CatalogHelper";
import { isEmptyResult } from "./helpers/errorHandlers";
import * as Model from "./helpers/model";
import { ICommonVisualizationProps } from "./_defunct/to_delete/VisualizationLoadingHOC";
import { ErrorComponent } from "./components/simple/ErrorComponent";
import { LoadingComponent } from "./components/simple/LoadingComponent";
// import { Kpi } from "./_defunct/kpi/Kpi";
// import { Visualization } from "./_defunct/uri/Visualization";
import { ErrorStates, ErrorCodes } from "./constants/errorStates";
import { VisualizationTypes, ChartType, VisualizationEnvironment } from "./constants/visualizationTypes";
// import { Execute } from "./execution/Execute";
import { IDrillableItem } from "./interfaces/DrillEvents";
import { IHeaderPredicate } from "./interfaces/HeaderPredicate";
import { IPushData, IColorsData } from "./interfaces/PushData";
import { AttributeFilter } from "./components/filters/AttributeFilter/AttributeFilter";
import { AttributeElements } from "./components/filters/AttributeFilter/AttributeElements";
import * as PropTypes from "./proptypes/index";
import { generateDimensions } from "./helpers/dimensions";
import * as BucketNames from "./constants/bucketNames";
import * as MeasureTitleHelper from "./helpers/measureTitleHelper";
import * as SortsHelper from "./helpers/sorts";
import DerivedMeasureTitleSuffixFactory from "./factory/DerivedMeasureTitleSuffixFactory";
import ArithmeticMeasureTitleFactory from "./factory/ArithmeticMeasureTitleFactory";
// import { IDataSourceProviderInjectedProps } from "./_defunct/afm/DataSourceProvider";

import { BarChart } from "./charts/barChart/BarChart";
import { ColumnChart } from "./charts/columnChart/ColumnChart";
import { LineChart } from "./charts/lineChart/LineChart";
import { AreaChart } from "./charts/areaChart/AreaChart";
import { PieChart } from "./charts/pieChart/PieChart";
import { Treemap } from "./charts/treemap/Treemap";
import { DonutChart } from "./charts/donutChart/DonutChart";
import { BubbleChart } from "./charts/bubbleChart/BubbleChart";
// import { PivotTable } from "./_defunct/pivotTable/PivotTable";
import { Headline } from "./charts/headline/Headline";
import { ScatterPlot } from "./charts/scatterPlot/ScatterPlot";
import { ComboChart } from "./charts/comboChart/ComboChart";
import { FunnelChart } from "./charts/funnelChart/FunnelChart";
import { Heatmap } from "./charts/heatmap/Heatmap";
import { withJsxExport } from "./components/withJsxExport";
import * as ChartConfiguration from "./interfaces/Config";
// tslint:disable-next-line:no-duplicate-imports
import { ILegendConfig, IChartConfig, IColorPalette, IColorPaletteItem } from "./interfaces/Config";
import Chart from "./components/visualizations/chart/Chart";
import ChartTransformation from "./components/visualizations/chart/ChartTransformation";
import { RuntimeError } from "./errors/RuntimeError";
import { IMeasureTitleProps, IArithmeticMeasureTitleProps } from "./interfaces/MeasureTitle";
import { OverTimeComparisonType, OverTimeComparisonTypes } from "./interfaces/OverTimeComparison";
import ColorUtils from "./components/visualizations/utils/color";
import * as HeaderPredicateFactory from "./factory/HeaderPredicateFactory";
import * as MappingHeader from "./interfaces/MappingHeader";
// import { BucketExecutor } from "./execution/BucketExecutor";

/**
 * CoreComponents
 * A collection of BaseChart, Headline, Table, ScatterPlot, FunnelChart
 * TODO: SDK8: revisit
 * @internal
 */
/*
const CoreComponents: ICoreComponents = {
    BaseChart: CoreBaseChart,
    Headline: CoreHeadline,
    Table: CoreTable,
    PivotTable: CorePivotTable,
    ScatterPlot: CoreScatterPlot,
    FunnelChart: CoreFunnelChart,
};
*/

export {
    AttributeElements,
    AttributeFilter,
    BarChart,
    BucketNames,
    CatalogHelper,
    Model,
    ChartType,
    ColumnChart,
    ScatterPlot,
    ComboChart,
    FunnelChart,
    ErrorCodes,
    ErrorStates,
    ErrorComponent,
    // Execute,
    // BucketExecutor,
    generateDimensions,
    Headline,
    ICommonVisualizationProps,
    IDrillableItem,
    ILegendConfig,
    IChartConfig,
    IColorPalette,
    IColorPaletteItem,
    IPushData,
    IColorsData,
    isEmptyResult,
    LoadingComponent,
    LineChart,
    AreaChart,
    PieChart,
    Treemap,
    BubbleChart,
    DonutChart,
    Heatmap,
    IMeasureTitleProps,
    IArithmeticMeasureTitleProps,
    MeasureTitleHelper,
    DerivedMeasureTitleSuffixFactory,
    ArithmeticMeasureTitleFactory,
    PropTypes,
    RuntimeError,
    VisEvents,
    VisualizationEnvironment,
    VisualizationTypes,
    ChartTransformation,
    Chart,
    OverTimeComparisonType,
    OverTimeComparisonTypes,
    SortsHelper,
    ChartConfiguration,
    ColorUtils,
    IHeaderPredicate,
    HeaderPredicateFactory,
    MappingHeader,
    withJsxExport,
};
