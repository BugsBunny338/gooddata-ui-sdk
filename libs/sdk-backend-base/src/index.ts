// (C) 2019-2020 GoodData Corporation

export {
    decoratedBackend,
    DecoratorFactories,
    DecoratedExecutionFactory,
    DecoratedExecutionResult,
    DecoratedPreparedExecution,
    PreparedExecutionWrapper,
} from "./decoratedBackend";
export { withEventing, AnalyticalBackendCallbacks } from "./eventingBackend";
export { dummyBackend, dummyBackendEmptyData, dummyDataView, dummyDataFacade } from "./dummyBackend";
