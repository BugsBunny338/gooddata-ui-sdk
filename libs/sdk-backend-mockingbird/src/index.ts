// (C) 2019-2020 GoodData Corporation

import {
    dummyBackend,
    dummyBackendEmptyData,
    dummyDataView,
    dummyDataFacade,
} from "@gooddata/sdk-backend-base";
export { dummyBackend, dummyBackendEmptyData, dummyDataView, dummyDataFacade };

export { recordedBackend } from "./recordedBackend";

export {
    RecordingIndex,
    ExecutionRecording,
    InsightRecording,
    DisplayFormRecording,
    ScenarioRecording,
    RecordedBackendConfig,
} from "./recordedBackend/types";

export {
    recordedDataView,
    DataViewAll,
    dataViewWindow,
    DataViewFirstPage,
} from "./recordedBackend/execution";

export {
    LegacyExecutionRecording,
    legacyRecordedBackend,
    legacyRecordedDataFacade,
} from "./legacyRecordedBackend";
