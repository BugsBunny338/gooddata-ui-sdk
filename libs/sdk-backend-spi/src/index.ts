// (C) 2019 GoodData Corporation

export {
    IAnalyticalBackend,
    IAnalyticalWorkspace,
    BackendCapabilities,
    AnalyticalBackendConfig,
    AnalyticalBackendFactory,
    prepareExecution,
} from "./backend";

export {
    IExecutionFactory,
    IPreparedExecution,
    IExecutionResult,
    IDataView,
    DimensionGenerator,
} from "./execution";

export {
    IExecutionDefinition,
    defFingerprint,
    defWithFilters,
    defSetSorts,
    defSetDimensions,
    emptyDef,
    newDefFromBuckets,
    newDefFromInsight,
    newDefFromItems,
} from "./execution/executionDefinition";

export { DataViewFacade } from "./execution/facade";

export {
    DataValue,
    IMeasureHeaderItem,
    IHeader,
    IResultDimension,
    IAttributeHeader,
    IMeasureGroupHeader,
    IResultAttributeHeaderItem,
    IResultHeaderItem,
    IResultMeasureHeaderItem,
    IResultTotalHeaderItem,
    ITotalHeaderItem,
    isAttributeHeader,
    isMeasureGroupHeader,
    isResultAttributeHeaderItem,
} from "./execution/results";

export {
    defForItems,
    defForBuckets,
    defForInsight,
    defWithDimensions,
    defWithSorting,
    defaultDimensionsGenerator,
} from "./execution/toolkit";

export { IFeatureFlagsQuery, IFeatureFlags } from "./featureFlags";

export { IWorkspaceMetadata } from "./metadata";

export { IElementQueryFactory, IElementQueryResult, IElementQuery, Element } from "./elements";

export { IExportConfig, IExportResult } from "./export";

export { IWorkspaceStyling } from "./styling";

export {
    AnalyticalBackendError,
    DataViewError,
    ExecutionError,
    NotSupported,
    NotImplemented,
    NotAuthenticated,
    isAnalyticalBackendError,
    isDataViewError,
    isExecutionError,
    isNotSupported,
    isNotImplemented,
    isNotAuthenticated,
} from "./errors";
