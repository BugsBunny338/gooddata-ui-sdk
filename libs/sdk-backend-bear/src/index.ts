// (C) 2019-2020 GoodData Corporation

import { AnalyticalBackendConfig, IAnalyticalBackend } from "@gooddata/sdk-backend-spi";
import { BearBackend, BearBackendConfig } from "./backend";
import { FixedLoginAndPasswordAuthProvider, ContextDeferredAuthProvider, BearAuthProviderBase } from "./auth";

/**
 * Returns function which creates instances of Analytical Backend implementation which works with the 'bear'
 * version of the GoodData platform.
 *
 * @param config - analytical backend configuration, may be omitted and provided later
 * @param implConfig - bear client specific configuration, may be omitted at this point but it cannot be provided later
 * @public
 */
function bearFactory(config?: AnalyticalBackendConfig, implConfig?: any): IAnalyticalBackend {
    return new BearBackend(config, implConfig);
}

export {
    BearBackendConfig,
    FixedLoginAndPasswordAuthProvider,
    ContextDeferredAuthProvider,
    BearAuthProviderBase,
};

export default bearFactory;

//
// Exports to support legacy state in AD
//

import { convertVisualization } from "./convertors/fromBackend/VisualizationConverter";

import { convertReferencesToUris } from "./convertors/fromBackend/ReferenceConverter";

import { convertInsight, convertInsightDefinition } from "./convertors/toBackend/InsightConverter";

import { toAfmExecution } from "./convertors/toBackend/afm/ExecutionConverter";

/**
 * Some of the convertors from and to bear types are exported through this so that they can be used by our
 * applications that were using bear-specific types in their state.
 *
 * All of these exports are marked as internal and can break at any time.
 *
 * @internal
 */
export const BearConvertors = {
    convertVisualization,
    convertReferencesToUris,
    convertInsight,
    convertInsightDefinition,
    toAfmExecution,
};
