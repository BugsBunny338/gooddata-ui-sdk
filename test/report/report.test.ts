// (C) 2007-2019 GoodData Corporation
import "isomorphic-fetch";
import fetchMock from "fetch-mock";
import { AFM, ExecuteAFM } from "@gooddata/typings";
import { ReportModule } from "../../src/report/report";
import { XhrModule, ApiResponseError } from "../../src/xhr";
import { IExportConfig, IExportResponse } from "../../src/interfaces";
import { mockPollingRequest, mockPollingRequestWithStatus } from "../helpers/polling";
import {
    SUCCESS_REQUEST_STATUS,
    ACCEPTED_REQUEST_STATUS,
    BAD_REQUEST_STATUS,
    BAD_REQUEST_MESSAGE,
    ERROR_RESTRICTED_MESSAGE,
} from "../../src/constants/errors";

const mockedReportModule = () => new ReportModule(new XhrModule(fetch, {}));

describe("report", () => {
    const createdReport = "/gdc/exporter/result/12345";
    const executionResult = "/executionResult/1234";
    const projectId = "1";
    const projectUri = `/gdc/internal/projects/${projectId}/exportResult`;

    describe("export", () => {
        afterEach(() => {
            fetchMock.restore();
        });

        describe("exportResult", () => {
            it("should sanitized showFilters config", () => {
                fetchMock.mock(projectUri, {
                    status: SUCCESS_REQUEST_STATUS,
                    body: { uri: createdReport },
                });

                const mockTask = (status: number) => ({ status, uri: createdReport });
                const finishedTask = mockTask(SUCCESS_REQUEST_STATUS);
                const runningTask = mockTask(ACCEPTED_REQUEST_STATUS);
                mockPollingRequest(createdReport, runningTask, finishedTask);

                const afm: AFM.IAfm = {
                    measures: [
                        {
                            localIdentifier: "a14fb77382304ab4925f05d2d64f7aed",
                            definition: {
                                measure: {
                                    item: {
                                        uri: "/gdc/md/projectId/obj/16203",
                                    },
                                },
                            },
                            alias: "Conversion",
                        },
                    ],
                    attributes: [
                        {
                            displayForm: {
                                uri: "/gdc/md/projectId/obj/15358",
                            },
                            localIdentifier: "645bf676a4854a32b694390bba9bd63c",
                        },
                    ],
                    filters: [
                        {
                            measureValueFilter: {
                                measure: {
                                    localIdentifier: "a14fb77382304ab4925f05d2d64f7aed",
                                },
                                condition: {
                                    comparison: {
                                        operator: "GREATER_THAN",
                                        value: 350000,
                                    },
                                },
                            },
                        },
                    ],
                };

                const expectedAfm: ExecuteAFM.IAfm = {
                    measures: [
                        {
                            localIdentifier: "a14fb77382304ab4925f05d2d64f7aed",
                            definition: {
                                measure: {
                                    item: {
                                        uri: "/gdc/md/projectId/obj/16203",
                                    },
                                },
                            },
                            alias: "Conversion",
                        },
                    ],
                    attributes: [
                        {
                            displayForm: {
                                uri: "/gdc/md/projectId/obj/15358",
                            },
                            localIdentifier: "645bf676a4854a32b694390bba9bd63c",
                        },
                    ],
                    filters: [
                        {
                            measureValueFilter: {
                                measure: {
                                    localIdentifier: "a14fb77382304ab4925f05d2d64f7aed",
                                },
                                condition: {
                                    comparison: {
                                        operator: "GREATER_THAN",
                                        value: 350000,
                                    },
                                },
                            },
                        },
                    ],
                };

                const exportConfig: IExportConfig = {
                    title: "title",
                    format: "xlsx",
                    mergeHeaders: false,
                    showFilters: true,
                    afm,
                };

                return mockedReportModule()
                    .exportResult(projectId, executionResult, exportConfig, { pollStep: 1 })
                    .then(() => {
                        const [, settings] = fetchMock.lastCall(
                            `/gdc/internal/projects/${projectId}/exportResult`,
                        );
                        expect(JSON.parse(settings.body as string)).toEqual({
                            resultExport: {
                                executionResult: "/executionResult/1234",
                                exportConfig: {
                                    title: "title",
                                    format: "xlsx",
                                    mergeHeaders: false,
                                    showFilters: true,
                                    afm: expectedAfm,
                                },
                            },
                        });
                    });
            });

            it("should return created file", () => {
                fetchMock.mock(projectUri, {
                    status: SUCCESS_REQUEST_STATUS,
                    body: { uri: createdReport },
                });

                const mockTask = (status: number) => ({ status, uri: createdReport });
                const finishedTask = mockTask(SUCCESS_REQUEST_STATUS);
                const runningTask = mockTask(ACCEPTED_REQUEST_STATUS);
                mockPollingRequest(createdReport, runningTask, finishedTask);

                const exportConfig: IExportConfig = {
                    title: "title",
                    format: "xlsx",
                    mergeHeaders: false,
                };

                return mockedReportModule()
                    .exportResult(projectId, executionResult, exportConfig, { pollStep: 1 })
                    .then((result: IExportResponse) => expect(result.uri).toEqual(createdReport));
            });

            it("should return error when polling fail", () => {
                const finishedTask = { status: BAD_REQUEST_STATUS };
                const runningTask = { status: ACCEPTED_REQUEST_STATUS, uri: createdReport };

                fetchMock.mock(projectUri, { uri: createdReport });

                mockPollingRequestWithStatus(createdReport, runningTask, finishedTask);

                const exportConfig: IExportConfig = {
                    title: "title",
                    format: "xlsx",
                    mergeHeaders: false,
                };

                return mockedReportModule()
                    .exportResult(projectId, executionResult, exportConfig, { pollStep: 1 })
                    .then(null, (error: ApiResponseError) => expect(error.message).toEqual("Bad Request"));
            });

            it("should return restricted error", () => {
                const finishedTask = {
                    status: BAD_REQUEST_STATUS,
                    // tslint:disable-next-line:max-line-length
                    body:
                        '{"error":{"message":"During export we\'ve detected user error: Export to required format is not allowed for data flagged as restricted."}}',
                };
                const runningTask = { status: ACCEPTED_REQUEST_STATUS, uri: createdReport };

                fetchMock.mock(projectUri, finishedTask);

                mockPollingRequest(createdReport, runningTask, finishedTask);

                const exportConfig: IExportConfig = {
                    title: "title",
                    format: "xlsx",
                    mergeHeaders: false,
                };

                return mockedReportModule()
                    .exportResult(projectId, executionResult, exportConfig, { pollStep: 1 })
                    .then(null, (error: ApiResponseError) => {
                        expect(error.response.status).toBe(BAD_REQUEST_STATUS);
                        expect(error.response.statusText).toBe(BAD_REQUEST_MESSAGE);
                        expect(error.message).toBe(ERROR_RESTRICTED_MESSAGE);
                    });
            });

            it("should reject with 400 when resource fails", () => {
                fetchMock.mock(projectUri, { status: BAD_REQUEST_STATUS });

                return mockedReportModule()
                    .exportResult(projectId, executionResult)
                    .then(null, (error: ApiResponseError) => {
                        // error thrown in xhr.ts
                        expect(error.response.status).toEqual(400);
                        expect(error.message).toEqual("Bad Request");
                    });
            });
        });
    });
});
