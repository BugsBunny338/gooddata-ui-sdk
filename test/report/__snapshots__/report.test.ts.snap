// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`report export exportResult should sanitize afm filter in export config 1`] = `
Object {
  "resultExport": Object {
    "executionResult": "/executionResult/1234",
    "exportConfig": Object {
      "afm": Object {
        "attributes": Array [
          Object {
            "displayForm": Object {
              "uri": "/gdc/md/projectId/obj/15358",
            },
            "localIdentifier": "645bf676a4854a32b694390bba9bd63c",
          },
        ],
        "filters": Array [
          Object {
            "positiveAttributeFilter": Object {
              "displayForm": Object {
                "uri": "bar",
              },
              "in": Object {
                "uris": Array [
                  "/gdc/md/bar1",
                  "/gdc/md/bar2",
                ],
              },
            },
          },
          Object {
            "negativeAttributeFilter": Object {
              "displayForm": Object {
                "identifier": "foo",
              },
              "notIn": Object {
                "values": Array [
                  "foo1",
                  "foo2",
                ],
              },
            },
          },
          Object {
            "absoluteDateFilter": Object {
              "dataSet": Object {
                "uri": "/gdc/md/i6k6sk4sznefv1kf0f2ls7jf8tm5ida6/obj/330",
              },
              "from": "2011-01-01",
              "to": "2011-12-31",
            },
          },
          Object {
            "relativeDateFilter": Object {
              "dataSet": Object {
                "uri": "/gdc/md/myproject/obj/921",
              },
              "from": -3,
              "granularity": "GDC.time.quarter",
              "to": 0,
            },
          },
          Object {
            "measureValueFilter": Object {
              "condition": Object {
                "comparison": Object {
                  "operator": "GREATER_THAN",
                  "value": 350000,
                },
              },
              "measure": Object {
                "localIdentifier": "a14fb77382304ab4925f05d2d64f7aed",
              },
            },
          },
        ],
        "measures": Array [
          Object {
            "alias": "Conversion",
            "definition": Object {
              "measure": Object {
                "item": Object {
                  "uri": "/gdc/md/projectId/obj/16203",
                },
              },
            },
            "localIdentifier": "a14fb77382304ab4925f05d2d64f7aed",
          },
        ],
      },
      "format": "xlsx",
      "mergeHeaders": false,
      "showFilters": true,
      "title": "title",
    },
  },
}
`;
